### PR TITLE
add last seen to the UserManager Api

### LINKED ISSUE #10550: Add field lastSeen in the API for the UsersManager
In order to better manage user-accounts for Piwik it would be great if the lastSeen variable was added as a new column that is returned with the API method UsersManager.getUsers. We currently use these API-calls to export the list of users and manage access to Piwik via an external tool.

### PR DESCRIPTION
**Pull request by peterhashair**
### Description:

add the last seen field to the UserManager.getUsers Api.

### File change
  - plugins/UsersManager/API.php ( add date_last_seen field to the api return) 
  - plugins/UsersManager/UserAccessFilter.php (it actualy loop users twice if not super admin. one in enrichUsers on in filterUsers)
  - plugins/UsersManager/tests/Integration/UsersManagerTest.php ( add test around the new field)
  - plugins/UsersManager/tests/System/ApiTest.php (accidently copied something there. Will remove)

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by peterhashair** [comment]
For the api date_last_seen maybe is not a propery word, if we addt ime onto as its value.  Suggest use last_seen if we want to use time, or we can remove the time, date_last_seen=date. Also for the time zone , should I use gmdate or just use date.

File: plugins/UsersManager/API.php [hunk 1]
```diff
@@ -879,6 +879,11 @@ private function enrichUser($user)
         unset($user['password']);
         unset($user['ts_password_modified']);
 
+        if($lastSeen = LastSeenTimeLogger::getLastSeenTimeForUser($user['login']))
+        {
```
Comment (sgiehl) [inline plugins/UsersManager/API.php]: minor coding style improvement, so it matches the other code
```suggestion
        if ($lastSeen = LastSeenTimeLogger::getLastSeenTimeForUser($user['login'])) {
```

File: plugins/UsersManager/tests/Integration/UsersManagerTest.php [hunk 2]
```diff
@@ -425,6 +425,7 @@ public function testGetUsersNoUser()
     }
 
     /**
+     * @todo
```
Comment (sgiehl) [inline plugins/UsersManager/tests/Integration/UsersManagerTest.php]: Is anything still left to do, or is this comment outdated?

File: plugins/UsersManager/API.php [hunk 3]
```diff
@@ -879,6 +879,11 @@ private function enrichUser($user)
...
+            $user['date_last_seen'] =  gmdate("Y-m-d\TH:i:s\Z", $lastSeen);
```
Comment (sgiehl) [inline plugins/UsersManager/API.php]: I guess we use to return the date format `Y-m-d H:i:s` in other places. See https://github.com/matomo-org/matomo/blob/c973567705a0065fdd7d7c7b11b80f1f0f1be350/core/Date.php#L389-L398

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: plugins/UsersManager/API.php [hunk 4]
```diff
@@ -879,6 +879,10 @@ private function enrichUser($user)
         unset($user['password']);
         unset($user['ts_password_modified']);
 
+        if ($lastSeen = LastSeenTimeLogger::getLastSeenTimeForUser($user['login'])) {
+            $user['last_seen'] = date("Y-m-d H:i:s", $lastSeen);
```
Comment (sgiehl) [inline plugins/UsersManager/API.php]: Guess we could use this here:
```suggestion
            $user['last_seen'] = Date::getDatetimeFromTimestamp($lastSeen);
```

**Review by sgiehl — APPROVED** [review]

**Pull request merged**